modern/src/RegisterDialog.js

```diff
@@ -0,0 +1,106 @@
+import t from './common/localization'
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import TextField from '@material-ui/core/TextField';
+import Snackbar from '@material-ui/core/Snackbar';
+
+const RegisterDialog = ({ showDialog, onResult }) => {
+  const [formFields, setFormFields] = useState({});
+  const [validationErrors, setValidationErrors] = useState({});
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+
+  const handleChange = (event) => {
+    setFormFields({ ...formFields, [event.target.name]: event.target.value });
+    setValidationErrors({ ...validationErrors, [event.target.name]: false });
+  }
+
+  const handleRegister = async (event) => {
+    let objErrors = {};
+    if (formFields.name.trim() === '') {
+      objErrors.name = true;
+    }
+    if (!/(.+)@(.+)\.(.{2,})/.test(formFields.email)) {
+      objErrors.email = true;
+    }
+    if (formFields.password.trim() === '') {
+      objErrors.password = true;
+    }
+    if (Object.keys(objErrors).length !== 0) {
+      setValidationErrors(objErrors);
+      return;
+    }
+
+    const response = await fetch('/api/users', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(formFields)
+    });
+
+    if (response.ok) {
+      showDialog = false;
+      setSnackbarOpen(true);
+    }
+  }
+
+  if (snackbarOpen) {
+    return <Snackbar
+      anchorOrigin={{ vertical: 'bottom', horizontal: 'center', }}
+      open={snackbarOpen} autoHideDuration={6000}
+      onClose={() => { onResult(true) }} message={t('loginCreated')}
+    />
+  } else if (showDialog) {
+    return <Dialog
+      open={true}
+      onClose={() => { onResult(false) }}>
+      <DialogContent>
+        <DialogContentText>{t('loginRegister')}</DialogContentText>
+        <TextField
+          margin='normal'
+          required
+          fullWidth
+          error={validationErrors.name}
+          label={t('sharedName')}
+          name='name'
+          value={formFields.name || ''}
+          autoComplete='name'
+          autoFocus
+          onChange={handleChange}
+          helperText={validationErrors.name && t('sharedRequired')} />
+        <TextField
+          margin='normal'
+          required
+          fullWidth
+          error={validationErrors.email}
+          label={t('userEmail')}
+          name='email'
+          value={formFields.email || ''}
+          autoComplete='email'
+          onChange={handleChange}
+          helperText={validationErrors.email && t('sharedRequired')} />
+        <TextField
+          margin='normal'
+          required
```

Comment:
Doesn't thins already handle "required"? Why do we need extra validation?

Reply:
It allowes all blanks to be entered as name and password. But maybe that's not important, so I removed validation (except e-mail).

Reply:
There is also a standard email validation, as far as I know. Can we use that?

Reply:
If you mean use type='email', seems this doesn't work with Dialog component (submit still works). I could disable submit button like this:  
disabled={!userName || !/(.+)@(.+)\.(.{2,})/.test(email) || !password}  ... let me know if that's ok.

Reply:
Yeah, let's do something like that, but maybe move that regex to a separate method for readability.